extensions/escape-game/standalone.mjs

```diff
@@ -0,0 +1,312 @@
+   * @return {Array}
+   */
+  async c2s_getListOfTerminals(player) {
+    return Array.from(this.terminals.keys());
+  }
+}
+
+/**
+ * The class to maintain the state of a room.
+ */
+class Room {
+  /**
+   * Create the standalone extension service object but does not start it.
+   * @constructor
+   * @param {string} roomId The identifier of the room.
+   * @param {} terminalServerGrpcService The gRPC service.
+   */
+  constructor(roomId, terminalServerGrpcService) {
+    this.roomId = roomId;
+    this.terminalServerGrpcService = terminalServerGrpcService;
+    this.players = [];
+    this.terminals = {}; // Mapping the terminal id to the container id in the terminal server.
+
+    this.defaultTerminals = [{imageName: 'debian:stable', terminalId: 'test'}];
+  }
+
+  /**
+   * Add a new player.
+   * @param {Player} player The player.
+   */
+  addPlayer(player) {
+    if (this.players.length >= MAX_PLAYER_PER_ROOM) {
+      return false;
+    }
+    return this.players.push(player);
+  }
+
+  /**
+   * Start all containers
+   */
+  async startContainers() {
+    for (const defaultTerminal of this.defaultTerminals) {
+      try {
+        let ret = await promisify(this.terminalServerGrpcService.CreateContainer.bind(this.terminalServerGrpcService))({
+          imageName: defaultTerminal.imageName
+        }, {deadline: new Date(Date.now() + 5000)});
+
+        if (!ret.success) {
+          console.error(`Fail to start container with image ${defaultTerminal.imageName}.`);
+          return false;
+        }
+        this.terminals[defaultTerminal.terminalId] = ret.containerId;
+      } catch (e) {
+        console.error(e);
+      }
+    }
+  }
+
+  /**
+   * Destroy the room.
+   */
+  async destroy() {
+    // Remove all containers
+    for (const defaultTerminal of this.defaultTerminals) {
+      try {
+        let ret = await promisify(this.terminalServerGrpcService.DestroyContainer.bind(this.terminalServerGrpcService))({
+          containerId: this.terminals[defaultTerminal.terminalId]
+        }, {deadline: new Date(Date.now() + 20000)});
+
+        if (!ret.success) {
+          console.error(`Fail to kill container ${defaultTerminal.terminalId} ${this.terminals[defaultTerminal.terminalId]}.`);
+          return false;
+        }
+      } catch (e) {
+        console.error(e);
+      }
+    }
+
+  }
+
+  /**
+   * Get JWT for the terminal server.
+   * @param {string} terminalId The identifier of the terminal.
+   */
+  getAccessToken(terminalId) {
+    if (!(terminalId in this.terminals)) {
+      throw new Error(`Terminal ${terminalId} doesn't exist.`);
+    }
+    return jwt.sign({
+      containerId: this.terminals[terminalId]
+    }, 'secret', {expiresIn: 10});
+   }
+}
+
+/**
+ * Terminal as an Interactive Object.
+ */
+class TerminalObject extends InteractiveObjectServerBaseClass {
+  async sf_showTerminal(playerID, kwargs) {
+    await this.helper.callS2cAPI(playerID, 'escape-game', 'showTerminalModal', 60*1000);
```

Comment:
I assume you'll need to tell the other side which terminal is being shown? Or better, include the token so we can save one round trip?